### ISSUE TITLE
Make Dependency Graph Searchable

### ISSUE THREAD

**Issue by frakman1** [fkie-cad/FACT_core#712]
The Dependency graph is nice but of limited use for large products. If I can see the module/library/executable, then I can click on it and follow the links. However with hundreds of circles, there's no way of finding the module I am looking for.

![image](https://user-images.githubusercontent.com/5826484/144525003-5fb33462-e5eb-4c8c-aa97-656ee87b8c5a.png)

Suggestion is to either allow CTL+F style text searching, or add a search widget that will highlight the circles that match the string being searched.

**Comment by dorpvom** [comment]
Cue:

![](https://boyter.org/static/books/a0f278b167ec23c35fe091f38096571a.jpg)

But for real: We'll look into this, but on first glance it seems to be more complicated than adding a few lines of code. But we might be able to add a search form. As non of us is a JS pro it might be a minute until we come around to do this though.

**Comment by rhelmke** [comment]
Hey,

so I am no web-pro, but I made an effort to re-implement the whole dependency graph functionality. You can find the code in pull request #725. It won't be merged before next January as most of us are already in the holidays. But feel free to check it out once it is merged into master.

Cheers
